src/app/machines/views/MachineList/MachineListTable/MachineListPagination/MachineListPagination.test.tsx

```diff
@@ -1,81 +1,124 @@
 import MachineListPagination, { Label } from "./MachineListPagination";
+import type { Props as MachineListPaginationProps } from "./MachineListPagination";
 
-import { render, screen } from "testing/utils";
-
-it("displays pagination if there are machines", () => {
-  render(
-    <MachineListPagination
-      currentPage={1}
-      itemsPerPage={20}
-      machineCount={100}
-      machinesLoading={false}
-      paginate={jest.fn()}
-    />
-  );
-  expect(
-    screen.getByRole("navigation", { name: Label.Pagination })
-  ).toBeInTheDocument();
-});
+import { fireEvent, render, screen, userEvent } from "testing/utils";
 
-it("does not display pagination if there are no machines", () => {
-  render(
-    <MachineListPagination
-      currentPage={1}
-      itemsPerPage={20}
-      machineCount={0}
-      machinesLoading={false}
-      paginate={jest.fn()}
-    />
-  );
-  expect(
-    screen.queryByRole("navigation", { name: Label.Pagination })
-  ).not.toBeInTheDocument();
-});
+describe("MachineListPagination - display", () => {
```

Comment:
Very good test coverage, but we haven't grouped tests in such way anywhere else ("display/functional").

I think it would be good to stick to existing conventions (and group tests simply by component or not group them at all which is what I prefer - as we normally have 1 test file per component anyway and such nesting isn't actually necessary) unless you feel strongly about this. In which case, let's discuss this.